### PR TITLE
Grid export issue  task 25963

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR will fix the customer_group column data is not reflected in the exported CSV 
### Fixed Issues (if relevant)

1. magento/magento2#25963: Grid Export rendered data is not reflecting in the exported File, Displayed ID instead of Rendered Label

### Manual testing scenarios (*)

1. Log in to the Magento Admin
2. Click "sales" from the menu on the left
3. Click "order"
4. Click on Export
5. Select CSV/Excel XML

### Expected Result
image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Sales/Ui/Component/DataProvider/Order/Document.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * Class Document
+ */
+class Document extends \Magento\Framework\View\Element\UiComponent\DataProvider\Document
+{
+
+    /**
+     * @var string
+     */
+    private static $customerGroupAttributeCode = 'customer_group';
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/Ui/Component/DataProvider/Order/Document.php:22]: Use `private const CUSTOMER_GROUP_ATTRIBUTE_CODE` instead.

File: app/code/Magento/Sales/Ui/Component/DataProvider/Order/Document.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+namespace Magento\Sales\Ui\Component\DataProvider\Order;
+
+use Magento\Customer\Api\GroupRepositoryInterface;
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class Document
+ */
+class Document extends \Magento\Framework\View\Element\UiComponent\DataProvider\Document
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/Ui/Component/DataProvider/Order/Document.php:16]: Import class with alias.

**Comment by konarshankar07** [comment]
I found that we can reuse the `Magento\Sales\Ui\Component\DataProvider\Document` class instead of creating new `Magento\Sales\Ui\Component\DataProvider\Order\Document` which is introduced in this PR because I think it violates DRY principle. We need to name the column for the `customer_group_id` needs to be same for whole application I found some inconsistency in `sales_order_grid` and `sales_invoice_grid` tables.

**Comment by novikor** [comment]
Hi, @konarshankar07. Thank you for your contribution.
The problem is reproduced globally while exporting any grid, even custom.

Could you please investigate for a more universal solution?
I am going to check it as well.

**Review by novikor — DISMISSED** [review]
I suggest taking a look at `\Magento\Framework\View\Element\UiComponent\DataProvider\SearchResult`.
Also `$document` can be moved to the constructor to create a possibility to provide it via `di.xml` for extensibility purposes.

Moreover, it is OK to modify `\Magento\Framework\View\Element\UiComponent\DataProvider\Document` directly if needed.

**Comment by konarshankar07** [comment]
@novikor ... Thanks for the feedback. I'll try to find the universal solution for this issue

**Comment by novikor** [comment]
@konarshankar07, - If we fail to find the universal solution, the current one is OK as well.
As far as I can see, similar problems for other grids are fixed this way.

Please let me know if you feel some trouble about that and would like to give up :)

**Comment by novikor** [comment]
Hi, @konarshankar07 !
Well, I dug too deep and created my own PR: https://github.com/magento/magento2/pull/26523

It seems that I cannot review this PR anymore since I am an interested person.
I`ll request @swnsma 's review.

**Comment by novikor** [comment]
Hi, @konarshankar07 . Please fix the static tests.

**Comment by konarshankar07** [comment]
Hello @novikor ,
Sorry for the delay. Sure, I'll fix the static test but did you find any universal solution for this issue?
Thanks

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
I've noticed few things to change

**Comment by novikor** [comment]
> Hello @novikor ,
> Sorry for the delay. Sure, I'll fix the static test but did you find any universal solution for this issue?

Hi. Yes, - it is https://github.com/magento/magento2/pull/26523
So, I afraid current PR is not needed anymore if my one is approved, - despite it is newer than yours.
@lbajsarowicz , what do you think?

**Pull request closed without merging**